ccp/src/prover.rs

```diff
@@ -14,129 +14,81 @@
  * limitations under the License.
  */
 
+mod cu_governor;
 #[cfg(test)]
 mod tests;
 
-use futures::future;
-use futures::FutureExt;
-use std::collections::HashMap;
 use std::sync::Arc;
-use std::sync::Mutex;
-use tokio::sync::oneshot;
-use tokio_util::sync::CancellationToken;
+use std::sync::Mutex as StdMutex;
 
 use ccp_config::CCPConfig;
 use ccp_msr::state::MSRState;
-use ccp_msr::{MSREnforce, MSRModeEnforcer};
+use ccp_msr::MSRModeEnforcer;
+use eyre::Context;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TryRecvError;
+use tokio::sync::oneshot;
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+use tokio::try_join;
+use tokio_util::sync::CancellationToken;
+
 use ccp_shared::nox_ccp_api::NoxCCPApi;
 use ccp_shared::proof::CCProof;
 use ccp_shared::proof::ProofIdx;
-use ccp_shared::types::*;
-use ccp_utils::run_utils::run_unordered;
+use ccp_shared::types::CUAllocation;
+use ccp_shared::types::EpochParameters;
 
-use crate::alignment_roadmap::*;
+use self::cu_governor::CUGovernor;
 use crate::cpuids_handle::CpuIdsHandle;
 use crate::cu::proving_thread::sync::to_state_operating_message::StateOperatingControlInlet;
 use crate::cu::proving_thread::sync::to_state_operating_message::StateOperatingControlMessage;
-use crate::cu::CUProver;
-use crate::cu::CUProverConfig;
-use crate::cu::CUResult;
 use crate::errors::CCProverError;
 use crate::hashrate::prometheus::PrometheusEndpoint;
 use crate::hashrate::HashrateCollector;
 use crate::hashrate::HashrateHandler;
 use crate::proof_storage::ProofStorageDrainer;
 use crate::state_operating_thread::StateOperatingThread;
 use crate::state_storage::CCPState;
+use crate::state_storage::Commitment;
 use crate::state_storage::StateStorage;
 use crate::stats_collecting_thread::StatsCollectingThread;
-use crate::status::CCStatus;
-use crate::status::ToCCStatus;
 
 const PROOF_DIR: &str = "cc_proofs";
 
 pub type CCResult<T> = Result<T, CCProverError>;
 
 pub struct CCProver {
-    cu_provers: HashMap<PhysicalCoreId, CUProver>,
-    cu_prover_config: CUProverConfig,
-    status: CCStatus,
+    governor_task: JoinHandle<()>,
+    governor_outlet: mpsc::Sender<GovernorMessage>,
```

Comment:
Is it worth to alias `mpsc::Sender<GovernorMessage>` as a new type for semantic purposes?

Reply:
Probably, yes.